### ISSUE TITLE
Shopping cart cleared automatically due to invalid product quantity

### ISSUE THREAD

**Issue by mohammed-ibenayad** [mohammed-ibenayad/exploratory_testing_opencart#3]
**Bug area** 
- Shopping Cart. 

**Describe the bug**
The system clears all items in the shopping cart while trying to update a product quantity with an invalid number (negative or floating-point number). 

**To Reproduce**
1. Add more than one product to the shopping cart.
2. View the shopping cart items.
3. Update the quantity of one item with a negative or floating-point number.
4. Click on the update button.

**Actual results**
The system deletes all items in the shopping cart and prompt the user to a screen showing a message that the shopping cart is empty.

**Test data**
NA

**Expected behavior**
The system should display an error stating that the product quantity should be an integer.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - OS: [Window 10 pro]
 - Browser [chrome]
 - Version [108]

**Smartphone (please complete the following information):**
NA

**Additional context**
NA

**Comment by mohammed-ibenayad** [comment]
you mean same actions to reproduce but different behavior noticed. #7 issue is about not updating the product quality, however in this issue, the problem is related to the side of effect which is deleting all items in the cart. So these are not similar bugs in my opinion.

**Comment by sojoodsweiti** [comment]
Yes, the same problem in the result but not the same behavior, in other words, another idea to test the functionality.
